### ISSUE TITLE
[DEVOPS] Deploy mobile

### ISSUE THREAD

**Issue by serv10** [unmsmfisi-socialapplication/social_app#396]
For the mobile deployment, I will use Firebase due to its seamless integration with essential mobile development services. Firebase App Distribution simplifies pre-release app sharing with testers, while Firebase Test Lab offers automated testing on diverse Android devices. Cloud Functions enable automation of deployment-related tasks, and real-time monitoring provides insights into app performance. 

In addition, with built-in authentication and security features, Firebase ensures secure access to deployment resources. It scales effectively and integrates with Google Cloud Platform, providing flexibility as your app grows. Supported by a thriving developer community and Google's reliability, Firebase streamlines CD, making it a favored choice among Android developers for its comprehensive suite of deployment and monitoring tools.

**Comment by serv10** [comment]
Here is an example of what will be implemented

``` yaml
name: Android CD with Firebase

on:
  push:
    branches:
      - main
    paths:
      - 'android/**'

defaults:
  run:
    working-directory: ./android

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout code
        uses: actions/checkout@v3

      - name: Set up Java
        uses: actions/setup-java@v3
        with:
          java-version: '17'
          distribution: 'adopt'
          cache: gradle

      - name: Build and Test
        run: |
          ./gradlew clean assembleDebug
          ./gradlew test

      - name: Deploy to Firebase App Distribution
        run: |
          firebase appdistribution:distribute \
            --token ${{ secrets.FIREBASE_TOKEN }} \
            --app <your-firebase-app-id> \
            --groups testers \
            --release-notes "New release notes" \
            --file app/build/outputs/apk/debug/app-debug.apk  # Replace with your APK path
        env:
          FIREBASE_TOKEN: ${{ secrets.FIREBASE_TOKEN }}
```

**Comment by serv10** [comment]
I've recently configured a continuous deployment pipeline using Firebase. There are a few minor differences from the setup described earlier. For example, instead of using **FIREBASE_TOKEN**, I opted for **GOOGLE_APPLICATION_CREDENTIALS_PATH**, which is the recommended approach today.

Other than that, the process remains quite similar. I've established a **MOBILE** group where I've added team members responsible for mobile development. This group receives an email containing the APK file each time a push is made to the develop branch.
